dashboard/app/controllers/registrations_controller.rb

```diff
@@ -223,7 +223,7 @@ def forbidden_change?(user, params)
   # ie if password or email was changed
   # extend this as needed
   def needs_password?(user, params)
-    return false if user.migrated? && user.encrypted_password.blank?
+    return false if user.migrated? && user.encrypted_password.blank? && params[:user][:password].blank?
```

Comment:
This comes up when you're adding a password, I assume?

Reply:
yep, i added this line when i did the "update email" feature for oauth users, but now that those users can also add a password, i needed to add an extra constraint